rotkehlchen/chain/evm/decoding/paraswap/decoder.py

```diff
@@ -98,33 +99,42 @@ def _decode_swap(
             log.error(f'Could not find the corresponding events when decoding paraswap swap {context.transaction.tx_hash.hex()}')  # noqa: E501
             return DEFAULT_DECODING_OUTPUT
 
-        if partial_refund_event is not None:  # if some in_asset is returned back
-            if in_event.asset == out_event.asset:  # check the similarity of asset with out_event
-                # due to order of events it can happen that this two get mislabeled and need to be swapped  # noqa: E501
-                partial_refund_event, in_event = in_event, partial_refund_event
+        if partial_refund_event is not None:  # if some in_asset is returned back.
+            # it's assumed in the above for loop, that the second in_event is the partial refund
+            # we verify below if the assumption is wrong, by checking the asset's similarity
+            if in_event.asset == out_event.asset:  # if this is true, then assumption is wrong
+                partial_refund_event, in_event = in_event, partial_refund_event  # swap them
+            # we check it here instead of in the for loop above because, out_event could be None
```

Comment:
```suggestion
            # it's assumed in the above for loop, that the second in_event is the partial refund
            # we verify below if the assumption is wrong, by checking the asset's similarity
            # we check it here instead of in the for loop above because, out_event could be None
            # at the time of the check
            if in_event.asset == out_event.asset:  # if this is true, then assumption is wrong
                partial_refund_event, in_event = in_event, partial_refund_event  # swap them
            
```